src/Data/Vector/Generic/Sized.hs

```diff
@@ -378,6 +376,18 @@ instance KnownNat n => Rep.Representable (Vector Boxed.Vector n) where
   index = Data.Vector.Generic.Sized.index
   {-# inline index #-}
 
+instance (Eq a, Hashable a) => Hashable (Vector Boxed.Vector n a) where
```

Comment:
This seems to different from the *vector-instances* instance, I think?  That library basically is `\salt -> hashWithSalt salt . toList`, which is https://github.com/tibbe/hashable/blob/9fa8af2eb38ed0b0b34c2da6631f877189ada7c0/Data/Hashable/Class.hs#L607-L611

I don't think there's anything wrong with this implementation here in particular, but we *might* be breaking user expectations by having `hash <$> toSized` be different from `Just . hash`.  And changing a `Hashable` instance is potentially a big breaking change in the future, so it might be better to fix this now when it's still easy than to wait for someone to come along with an issue.

Reply:
Actually this instance might be problematic because it hashes the empty vector to the salt itself.  One can then determine the salt being used by feeding it the empty vector.

Reply:
That's a good point. I'm gonna fix that one to match vector-instances, for this exact reason. Didn't realize you could do a 'desalination attack' like that!